source/MissionAction.cpp

```diff
@@ -306,7 +306,7 @@ void MissionAction::Do(PlayerInfo &player, UI *ui, const System *destination,
 		subs["<last>"] = player.LastName();
 		if(player.Flagship())
 			subs["<ship>"] = player.Flagship()->Name();
-		string text = Format::Replace(dialogText, subs);
+		string text = Format::Replace(Phrase::ExpandPhrases(dialogText), subs);
```

Comment:
I don't think you need to do this here if it's already done in `MissionAction::Instantiate`.